### DISCUSSION TITLE
Grade Change & Bonus Scoring Ideas

### DISCUSSION THREAD

**Discussion by hburn7** [ppy/osu#18954] [Ideas]
This idea is about adding grades to the game, in addition to using a behind-the-scenes OD11 calculation for rewarding bonus score. This idea is very experimental and not fully thought out in my mind but I think could make a great addition to the game if done properly.

Currently, scoring and grading are not tightly correlated. It is possible to achieve a very low scoring S rank, similarly it is possible to get a high scoring C rank. I have seen #14757 and #14078 and agree with the 50/50 split as osu! is just as much of an accuracy game as it is a combo game, in my opinion. osu! is about clicking the circles, _to the beat [accurately]_, after all.

My idea is to add more variance in terms of grading. Right now, it feels lackluster to not have any grade differential between a 95% S rank with misses and a 99.9% S rank full-combo. An SS is only given for a 100% accuracy score _on the current OD_. Adding more grades, such as a triple-S (SSS) and triple/double grades for A and B (i.e. AAA/AA, BBB/BB) feels like it could be a good addition to the game by adding more diversity to achievable scores and serves as a better way to compare scores quickly. The triple/double grades for A and B grades are a less thought out idea and I'm not sure of the details there myself. However, it feels as though there is a lot lacking between S and SS grades right now.

A rough idea for the AAA/AA & BBB/BB grades could be based purely on accuracy achieved with misses. Given Lazer's current direction with S ranks being rewarded despite having misses, though, I don't believe this would work. If the 50/50 accuracy/combo score split were to be implemented, more grades could be introduced based purely on score. So a SS could be 1,000,000 score (osu!Standard), S could be 950,000 score, something like that - take the numbers with a grain of salt.

Adding a triple-S grade makes a lot of sense to me and here's how it would work:
- SSS grade only rewarded for a 100% accuracy SS through a hidden OD11 calculation
- SSS grade achieved with any score that has 100% accuracy with OD11 applied.
- osu!Mania perfect 1,000,000 score.

In addition to the above, adding a hidden OD11 calculation can help differentiate score ties on leaderboards:
- For every perfect note hit within the OD11 timing window, +1 score is rewarded

Currently, I am unfamiliar with the way score bonus works outside of spinners for SS plays - but I have noticed that while achieving SS plays I have gotten bonus score.

Some scores are worthy of being given an extra mark of perfection. Achieving an SS on any map is difficult, but rewarding a perfect OD11 SS / o!Mania 1,000,000 with extra prestige makes perfect sense to me.

None of the above outside of grades being given based on score solves the S - SS gap issue but I'm curious to what everyone thinks about these ideas.

**Comment by iSimpFor** [comment]
i don't think there should be so many grades but a few can be added, maybe 1-2 grades after the current S-rank at least

i think the bonus points for hitting in the OD11 timing window shouldn't really be hidden but it should have less emphasis to hit compared to the other judgements

maybe something like "PERFECT 854+86"/"GREAT 854+86" where the left number is regular judgements and the right number is the OD11 judgements 

if one manages to hit all OD11 perfects it would something like "PERFECT +1902"/"GREAT +1902" with nothing on the left
